### ISSUE TITLE
Pickle error after latest update

### ISSUE THREAD

**Issue by xBlackPoison357x** [magicbear/palworld-server-toolkit#56]
Updated to the latest version and getting this error when attempting to load up a Level.sav, didn't do it before I updated.

`
C:\Users\trexl\Desktop\Palworld 2>python -m palworld_server_toolkit.editor Level.sav
Loading Level.sav...Done in 1.48s.
Parsing Level.sav...Decodeing Failed on Decodeing Path .worldSaveData -> <class 'TypeError'>: cannot pickle 'weakref.ReferenceType' object
Decodeing Failed on Decodeing Path  -> <class 'TypeError'>: cannot pickle 'weakref.ReferenceType' object
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\trexl\Desktop\Palworld 2\palworld_server_toolkit\editor.py", line 4752, in <module>
    main()
  File "C:\Users\trexl\Desktop\Palworld 2\palworld_server_toolkit\editor.py", line 514, in main
    LoadFile(args.filename)
  File "C:\Users\trexl\Desktop\Palworld 2\palworld_server_toolkit\editor.py", line 2213, in LoadFile
    gvas_file = ProgressGvasFile.read(raw_gvas, PALWORLD_TYPE_HINTS, SKP_PALWORLD_CUSTOM_PROPERTIES)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\trexl\Desktop\Palworld 2\palworld_server_toolkit\editor.py", line 264, in read
    gvas_file.properties = reader.properties_until_end()
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\trexl\Desktop\Palworld 2\palworld_server_toolkit\palobject.py", line 1047, in properties_until_end
    raise e
  File "C:\Users\trexl\Desktop\Palworld 2\palworld_server_toolkit\palobject.py", line 1044, in properties_until_end
    properties[name] = self.property(type_name, size, f"{path}.{name}")
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\trexl\Desktop\Palworld 2\palworld_server_toolkit\palobject.py", line 991, in property
    return super().property(type_name, size, path, nested_caller_path)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\trexl\Desktop\Palworld 2\palworld_server_toolkit\..\palworld_save_tools\archive.py", line 304, in property
    value = self.struct(path)
            ^^^^^^^^^^^^^^^^^
  File "C:\Users\trexl\Desktop\Palworld 2\palworld_server_toolkit\..\palworld_save_tools\archive.py", line 415, in struct
    value = self.struct_value(struct_type, path)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\trexl\Desktop\Palworld 2\palworld_server_toolkit\..\palworld_save_tools\archive.py", line 442, in struct_value
    return self.properties_until_end(path)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\trexl\Desktop\Palworld 2\palworld_server_toolkit\palobject.py", line 1047, in properties_until_end
    raise e
  File "C:\Users\trexl\Desktop\Palworld 2\palworld_server_toolkit\palobject.py", line 1042, in properties_until_end
    self.processlist[sub_path] = self.load_mp_array(properties[name], sub_path, size)
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\trexl\Desktop\Palworld 2\palworld_server_toolkit\palobject.py", line 931, in load_mp_array
    p.start()
  File "C:\Users\trexl\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\process.py", line 121, in start
    self._popen = self._Popen(self)
                  ^^^^^^^^^^^^^^^^^
  File "C:\Users\trexl\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\context.py", line 224, in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\trexl\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\context.py", line 337, in _Popen
    return Popen(process_obj)
           ^^^^^^^^^^^^^^^^^^
  File "C:\Users\trexl\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\popen_spawn_win32.py", line 95, in __init__
    reduction.dump(process_obj, to_child)
  File "C:\Users\trexl\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\reduction.py", line 60, in dump
    ForkingPickler(file, protocol).dump(obj)
TypeError: cannot pickle 'weakref.ReferenceType' object
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Users\trexl\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\spawn.py", line 122, in spawn_main
    exitcode = _main(fd, parent_sentinel)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\trexl\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\spawn.py", line 132, in _main
    self = reduction.pickle.load(from_parent)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
EOFError: Ran out of input
Exception ignored in: <function SharedMemory.__del__ at 0x000001F4051D4040>
Traceback (most recent call last):
  File "C:\Users\trexl\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\shared_memory.py", line 187, in __del__
    self.close()
  File "C:\Users\trexl\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\shared_memory.py", line 230, in close    self._mmap.close()
BufferError: cannot close exported pointers exist
Exception ignored in: <function SharedMemory.__del__ at 0x000001F4051D4040>
Traceback (most recent call last):
  File "C:\Users\trexl\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\shared_memory.py", line 187, in __del__
  File "C:\Users\trexl\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\shared_memory.py", line 230, in closeBufferError: cannot close exported pointers exist`

**Comment by magicbear** [comment]
can u provide the sav file?

**Comment by healla** [comment]
I get same error on 0.7.2 had to install 0.6.9 again

**Comment by wcdfilll** [comment]
> bạn có thể cung cấp file sav được không?
[83AEF92D42D388ED7C3383AA3D0E05AC.zip](https://github.com/magicbear/palworld-server-toolkit/files/14312709/83AEF92D42D388ED7C3383AA3D0E05AC.zip)

**Comment by jkroepke** [comment]
Having same errors:

```
 python -m palworld_server_toolkit.editor .\6EE886BC41FB3023717E0D9DB323E1C5\Level.sav
Loading .\6EE886BC41FB3023717E0D9DB323E1C5\Level.sav...Done in 0.24s.
Parsing .\6EE886BC41FB3023717E0D9DB323E1C5\Level.sav...Decodeing Failed on Decodeing Path .worldSaveData -> <class 'TypeError'>: cannot pickle 'weakref.ReferenceType' object
Decodeing Failed on Decodeing Path  -> <class 'TypeError'>: cannot pickle 'weakref.ReferenceType' object
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\Jan\AppData\Local\Programs\Python\Python312\Lib\site-packages\palworld_server_toolkit\editor.py", line 4757, in <module>
    main()
  File "C:\Users\Jan\AppData\Local\Programs\Python\Python312\Lib\site-packages\palworld_server_toolkit\editor.py", line 514, in main
    LoadFile(args.filename)
  File "C:\Users\Jan\AppData\Local\Programs\Python\Python312\Lib\site-packages\palworld_server_toolkit\editor.py", line 2218, in LoadFile
    gvas_file = ProgressGvasFile.read(raw_gvas, PALWORLD_TYPE_HINTS, SKP_PALWORLD_CUSTOM_PROPERTIES)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Jan\AppData\Local\Programs\Python\Python312\Lib\site-packages\palworld_server_toolkit\editor.py", line 264, in read
    gvas_file.properties = reader.properties_until_end()
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Jan\AppData\Local\Programs\Python\Python312\Lib\site-packages\palworld_server_toolkit\palobject.py", line 1046, in properties_until_end
    raise e
  File "C:\Users\Jan\AppData\Local\Programs\Python\Python312\Lib\site-packages\palworld_server_toolkit\palobject.py", line 1043, in properties_until_end
    properties[name] = self.property(type_name, size, f"{path}.{name}")
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Jan\AppData\Local\Programs\Python\Python312\Lib\site-packages\palworld_server_toolkit\palobject.py", line 990, in property
    return super().property(type_name, size, path, nested_caller_path)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Jan\AppData\Local\Programs\Python\Python312\Lib\site-packages\palworld_server_toolkit\..\palworld_save_tools\archive.py", line 304, in property
    value = self.struct(path)
            ^^^^^^^^^^^^^^^^^
  File "C:\Users\Jan\AppData\Local\Programs\Python\Python312\Lib\site-packages\palworld_server_toolkit\..\palworld_save_tools\archive.py", line 415, in struct
    value = self.struct_value(struct_type, path)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Jan\AppData\Local\Programs\Python\Python312\Lib\site-packages\palworld_server_toolkit\..\palworld_save_tools\archive.py", line 442, in struct_value
    return self.properties_until_end(path)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Jan\AppData\Local\Programs\Python\Python312\Lib\site-packages\palworld_server_toolkit\palobject.py", line 1046, in properties_until_end
    raise e
  File "C:\Users\Jan\AppData\Local\Programs\Python\Python312\Lib\site-packages\palworld_server_toolkit\palobject.py", line 1038, in properties_until_end
    self.processlist[sub_path] = self.load_mp_map(properties[name], sub_path, size)
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Jan\AppData\Local\Programs\Python\Python312\Lib\site-packages\palworld_server_toolkit\palobject.py", line 893, in load_mp_map
    p.start()
  File "C:\Users\Jan\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\process.py", line 121, in start
    self._popen = self._Popen(self)
                  ^^^^^^^^^^^^^^^^^
  File "C:\Users\Jan\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\context.py", line 224, in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Jan\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\context.py", line 337, in _Popen
    return Popen(process_obj)
           ^^^^^^^^^^^^^^^^^^
  File "C:\Users\Jan\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\popen_spawn_win32.py", line 95, in __init__
    reduction.dump(process_obj, to_child)
  File "C:\Users\Jan\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\reduction.py", line 60, in dump
    ForkingPickler(file, protocol).dump(obj)
TypeError: cannot pickle 'weakref.ReferenceType' object
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Users\Jan\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\spawn.py", line 122, in spawn_main
    exitcode = _main(fd, parent_sentinel)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Jan\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\spawn.py", line 132, in _main
    self = reduction.pickle.load(from_parent)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
EOFError: Ran out of input
Exception ignored in: <function SharedMemory.__del__ at 0x000002773F83C720>
Traceback (most recent call last):
  File "C:\Users\Jan\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\shared_memory.py", line 187, in __del__    self.close()
  File "C:\Users\Jan\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\shared_memory.py", line 230, in close
    self._mmap.close()
BufferError: cannot close exported pointers exist
Exception ignored in: <function SharedMemory.__del__ at 0x000002773F83C720>
Traceback (most recent call last):
  File "C:\Users\Jan\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\shared_memory.py", line 187, in __del__  File "C:\Users\Jan\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\shared_memory.py", line 230, in close
BufferError: cannot close exported pointers exist
```
[6EE886BC41FB3023717E0D9DB323E1C5.zip](https://github.com/magicbear/palworld-server-toolkit/files/14314544/6EE886BC41FB3023717E0D9DB323E1C5.zip)

**Comment by magicbear** [comment]
fixed on v0.7.2
